src/models.ts

```diff
@@ -297,3 +304,33 @@ export class AdvancedFilter extends Filter {
     return filter;
   }
 }
+
+export interface IDataReference {
+  target: IFilterTarget;
+}
+
+export interface IEqualsDataReference extends IDataReference {
+  equals: string | boolean | number;
+}
+
+export interface IBetweenDataReference extends IDataReference {
+  between: (string | boolean | number)[];
+}
+
+export interface IValueDataReference extends IDataReference {
+  value: string | boolean | number;
```

Comment:
It's optional because it's just not available on the client (we would need to run a query against the service to retrieve that actual value. I think on the client there's only a formatted string value).

But in general both approaches have some pros and cons. I don't have a strong opinion here, with can go with required.